lib/services/init-service.ts

```diff
@@ -0,0 +1,125 @@
+import Future = require("fibers/future");
+import path = require("path");
+import semver = require("semver");
+
+export class InitService implements IInitService {
+	private static MIN_SUPPORTED_FRAMEWORK_VERSIONS: IStringDictionary = {
+		"tns-ios": "1.1.0",
+		"tns-android": "1.1.0"	
+	};
+	
+	private _projectFilePath: string;
+	
+	constructor(private $fs: IFileSystem,
+		private $errors: IErrors,
+		private $logger: ILogger,
+		private $options: IOptions,
+		private $injector: IInjector,
+		private $staticConfig: IStaticConfig,
+		private $projectHelper: IProjectHelper,
+		private $prompter: IPrompter,
+		private $npm: INodePackageManager,
+		private $npmInstallationManager: INpmInstallationManager) { }
+	
+	public initialize(): IFuture<void> {
+		return (() => {
+			let projectData: any = { };
+			
+			if(this.$fs.exists(this.projectFilePath).wait()) {
+				projectData = this.$fs.readJson(this.projectFilePath).wait();
+			}
+			
+			let projectDataBackup = _.extend({}, projectData);			
+			
+			if(!projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE]) {
+				projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE] = { };
+				this.$fs.writeJson(this.projectFilePath, projectData).wait(); // We need to create package.json file here in order to prevent "No project found at or above and neither was a --path specified." when resolving platformsData				
+			}
+			
+			try {
+				
+				projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE]["id"] = this.getProjectId().wait();
+				
+				if(this.$options.frameworkName && this.$options.frameworkVersion) {
+					projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE][this.$options.frameworkName] = this.buildVersionData(this.$options.frameworkVersion);
+				} else {
+					let $platformsData = this.$injector.resolve("platformsData");
+					_.each($platformsData.platformsNames, platform => {
+						let platformData: IPlatformData = $platformsData.getPlatformData(platform);
+						if(!platformData.targetedOS || (platformData.targetedOS && _.contains(platformData.targetedOS, process.platform))) {
+							projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE][platformData.frameworkPackageName] = this.getVersionData(platformData.frameworkPackageName).wait();
+						}
+					});
+				}
+				
+				this.$fs.writeJson(this.projectFilePath, projectData).wait();
+			} catch(err) {
+				this.$fs.writeJson(this.projectFilePath, projectDataBackup).wait();
+				throw err;
+			}
+			
+			this.$logger.out("Project successfully initialized."); 
+		}).future<void>()();
+	}
+	
+	private get projectFilePath(): string {
+		if(!this._projectFilePath) {
+			let projectPath = path.resolve(this.$options.path || ".");
+			this._projectFilePath = path.join(projectPath, constants.PACKAGE_JSON_FILE_NAME);
+		}
+		
+		return this._projectFilePath;
+	}
+	
+	private set projectFilePath(newProjectFilePath: string) {
+		this._projectFilePath = newProjectFilePath;
```

Comment:
Should we verify that newProjectFilePath is correct? I.e. is non-empty string and possibly that exists on the fs?